plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/RemoteQueryCancellationModule.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import com.google.inject.Binder;
+import com.google.inject.Key;
+import com.google.inject.Module;
+import com.google.inject.Scopes;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.trino.plugin.base.CatalogName;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Qualifier;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.util.concurrent.ExecutorService;
+
+import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static io.airlift.configuration.ConditionalModule.installModuleIf;
+import static java.lang.String.format;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newCachedThreadPool;
+
+public class RemoteQueryCancellationModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        install(installModuleIf(
+                RemoteQueryCancellationConfig.class,
+                RemoteQueryCancellationConfig::isRemoteQueryCancellationEnabled,
+                bindForRecordCursor()));
+    }
+
+    private static Module bindForRecordCursor()
+    {
+        return binder -> {
+            newOptionalBinder(binder, Key.get(ExecutorService.class, ForRecordCursor.class))
+                    .setBinding()
+                    .toProvider(CachedThreadPoolProvider.class)
+                    .in(Scopes.SINGLETON);
+            binder.bind(String.class).annotatedWith(CachedThreadPoolProvider.Name.class).toInstance("record-cursor");
```

Comment:
redundant (we're in a code that's specific to record cursor's query cancellation)
& does seem to make the class reusable -- you still can bind it once only (actually: you can reuse class in separate guice modules, but that's unlikely to be useful in the context of base-jdbc)

perhaps @losipiuk can suggest how to make it more future-proof so that we could bind `CachedThreadPoolProvider` multiple times, with different thread pool name patterns